compiler/rustc_mir/src/borrow_check/diagnostics/mod.rs

```diff
@@ -630,14 +644,35 @@ impl UseSpans<'_> {
         }
     }
 
+    // Add a span label to the use of the captured variable, if it exists.
+    // only adds label to the `path_span`
+    pub(super) fn var_span_label_path_only(
```

Comment:
Why separate this out?

Reply:
We don't always need both pieces of information. e.g. 
```rust
fn main() {
    let mut c = {
        let mut p = Point {x: "1".to_string(), y: "2".to_string() };
        || {
           let x = &mut p.x;
           println!("{:?}", p);
            //~^ ERROR `p` does not live long enough
        }
    };
    c();
}
```
Here, we really only care that p is borrowed, not that it is a mutable borrow or why it is mutable, so just showing path span should be fine. This shuold allow for better signal to noise ratio in some messages.

Reply:
I see